### PR TITLE
WT-8570 WT-8702 WT-8709 WT-8825 WT-11492 WT-9248 (v4.4 backport)

### PR DESCRIPTION
**Pull request by etienneptl**
The backports are done together due to their dependency:
[WT-8570](https://jira.mongodb.org/browse/WT-8570)
[WT-8702](https://jira.mongodb.org/browse/WT-8702) (caused by [WT-8570](https://jira.mongodb.org/browse/WT-8570))
[WT-8709](https://jira.mongodb.org/browse/WT-8709) (caused by [WT-8702](https://jira.mongodb.org/browse/WT-8702))

WT-8825, WT-9248 and WT-11492 have also been included as they create/modify/fix the new test `test_checkpoint_snapshot05.py`.

### CODE REVIEW

**Review by mm-ng — APPROVED** [review]
LGTM

**Comment by etienneptl** [comment]
Despite the additional backports, test_checkpoint_snapshot05 still fails.

**Comment by lukech** [comment]
There's a test failure in the PR patch build. Could it be related to one of the tickets being cherry-picked?

**Comment by etienneptl** [comment]
> There's a test failure in the PR patch build. Could it be related to one of the tickets being cherry-picked?

Yes, @lukech. It seems that this test has been changed by multiple tickets, I have added one more ticket to the backport: WT-9248. Hopefully, this was the last one missing.

**Comment by etienneptl** [comment]
I think I am understanding the issue, CS scenarios are not supported in 4.4.

**Comment by kommiharibabu** [comment]
@etienneptl As the test is not stable, how about not backporting the test into 4.4 branch and backport only the code changes. This will avoid backporting extra changes.

**Comment by etienneptl** [comment]
> @etienneptl As the test is not stable, how about not backporting the test into 4.4 branch and backport only the code changes. This will avoid backporting extra changes.

@kommiharibabu, I see that this would limit the number of tickets and code changes for this backport. However, it would still be better to have a way to test the new code. I am happy to get extra feedback, @lukech, what do you think?

File: test/suite/test_checkpoint_snapshot05.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            if self.value_format == '8t':
+                cursor1.set_value(self.valueb)
+            else:
+                cursor1.set_value(self.valueb + str(i))
+            cursor1.set_value(self.valueb)
+            self.assertEqual(cursor1.update(), 0)
+
+        # Create a checkpoint thread
+        done = threading.Event()
+        ckpt = checkpoint_thread(self.conn, done)
```
Comment (marcbutler) [inline test/suite/test_checkpoint_snapshot05.py:147]: This does not have the changes from WT-11492. Is that intentional?

Comment (etienneptl) [reply test/suite/test_checkpoint_snapshot05.py:147]: I must have missed them, sorry. I will double check everything, thank you for catching this!

Comment (marcbutler) [reply test/suite/test_checkpoint_snapshot05.py:147]: You should see modifications to `wtthread.py` and this test. I'll keep an eye out when you push (or you can ping me if you need quick turnaround) and can verify for you almost instantly.

**Comment by lukech** [comment]
@etienneptl @kommiharibabu I'd prefer we disable the column-store part of the test and leave the row-store tested if possible.

**Comment by marcbutler** [comment]
> @etienneptl @kommiharibabu I'd prefer we disable the column-store part of the test and leave the row-store tested if possible.

The pr in it's current state appears to be missing WT-11492 which addresses that test.

**Comment by etienneptl** [comment]
> > @etienneptl @kommiharibabu I'd prefer we disable the column-store part of the test and leave the row-store tested if possible.
> 
> The pr in it's current state appears to be missing [WT-11492](https://jira.mongodb.org/browse/WT-11492) which addresses that test.

The [force push](https://github.com/wiredtiger/wiredtiger/compare/937ab214321ad4ab1da289f04edeff0295999995..49edfdf65a79f7253c4c2928682d4e05b3f0ba56) I did completely remove this ticket. I must have executed the wrong git commands, I will rework this backport correctly and ping everyone.

**Comment by etienneptl** [comment]
@marcbutler, I have pushed again the changes which now contain WT-11492, this is the history:
```
    WT-8570 Do not increase oldest ID during recovery (#7438)
    WT-8702 Limit oldest ID to recovered checkpoint snapshot in recovery (#7447)
    WT-8709 Do not hold the oldest id of logged tables (#7490) (#9140)
    WT-8825 Fix test_checkpoint_snapshot05 expected value validation failure (#9063) (#9177) (#9257)
    WT-11492 Fix false positive failure in test_checkpoint_snapshot05 (#9587)
    WT-9248 Fix test_checkpoint_snapshot failure and convert python tests to use stats logic wait rather than sleep (#7898)
```

**Comment by marcbutler** [comment]
> @marcbutler, I have pushed again the changes which now contain [WT-11492](https://jira.mongodb.org/browse/WT-11492), this is the history:
> 
> ```
>     WT-8570 Do not increase oldest ID during recovery (#7438)
>     WT-8702 Limit oldest ID to recovered checkpoint snapshot in recovery (#7447)
>     WT-8709 Do not hold the oldest id of logged tables (#7490) (#9140)
>     WT-8825 Fix test_checkpoint_snapshot05 expected value validation failure (#9063) (#9177) (#9257)
>     WT-11492 Fix false positive failure in test_checkpoint_snapshot05 (#9587)
>     WT-9248 Fix test_checkpoint_snapshot failure and convert python tests to use stats logic wait rather than sleep (#7898)
> ```

I can see the changes from WT-11492 in the diff now.

**Comment by etienneptl** [comment]
> @etienneptl `test_snapshot05` is failing but the fix is clearly in the patch, so will require debugging.

Yes, I needed to disable the CS scenarios. It is now done.

**Review by lukech — APPROVED** [review]

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by clarissecheah — APPROVED** [review]
LGTM

**Review by marcbutler — APPROVED** [review]
lgtm

**Pull request merged**